### ISSUE TITLE
Feature Request : Stacked expandable notifications

### ISSUE THREAD

**Issue by HosamHasanRamadan** [payam-zahedi/toastification#88]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
Stacked expandable notifications  

**Describe alternatives you've considered**
- https://pub.dev/packages/delightful_toast
- https://pub.dev/packages/stacked_notification_cards
- https://github.com/dokar3/compose-sonner - [example](https://dokar3.github.io/compose-sonner/)

**Additional context**
<video src="https://private-user-images.githubusercontent.com/50796600/246908436-59b95cb7-9068-4f3e-8469-0b35d9de5cf0.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MjQ4NzIsIm5iZiI6MTcwNjYyNDU3MiwicGF0aCI6Ii81MDc5NjYwMC8yNDY5MDg0MzYtNTliOTVjYjctOTA2OC00ZjNlLTg0NjktMGIzNWQ5ZGU1Y2YwLm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMwVDE0MjI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNmRiNzlkOTA5NTcyZmFmMjI0NGI2YmYxOGYwMGQ4NWE3MmFiZGRmOTc1MDRkYmYwZmFiZTMxZDIzNzM4OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TpZGEuxshurqtTCuOiJdMMnfIGWJ3_iroj0Bg7cflBg" data-canonical-src="https://private-user-images.githubusercontent.com/50796600/246908436-59b95cb7-9068-4f3e-8469-0b35d9de5cf0.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MjQ4NzIsIm5iZiI6MTcwNjYyNDU3MiwicGF0aCI6Ii81MDc5NjYwMC8yNDY5MDg0MzYtNTliOTVjYjctOTA2OC00ZjNlLTg0NjktMGIzNWQ5ZGU1Y2YwLm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTMwVDE0MjI1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzNmRiNzlkOTA5NTcyZmFmMjI0NGI2YmYxOGYwMGQ4NWE3MmFiZGRmOTc1MDRkYmYwZmFiZTMxZDIzNzM4OGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TpZGEuxshurqtTCuOiJdMMnfIGWJ3_iroj0Bg7cflBg" controls="controls" muted="muted" class="d-block rounded-bottom-2 border-top width-fit" style="max-height:640px; min-height: 200px">

  </video>

**Comment by payam-zahedi** [comment]
Thanks @HosamHasanRamadan for opening this issue.
For sure, this is a great Idea.

Let's see how we can achieve it in the future

**Comment by sagnik-sanyal** [comment]
Yeah it is a great feature request, would love to have it in near future. I guess we can now think about the implementations and ways on how to achieve this

**Comment by deepraj02** [comment]
Hey, this seems to be a good issue to work on, is someone currently working on it? 
I'd like to give it a shot

**Comment by payam-zahedi** [comment]
Hey @deepraj02 
I appreciate it 

It's yours now

**Comment by AntonioAEMartins** [comment]
@deepraj02, I wanted to follow up and ask if you've made any progress on this.

**Comment by AntonioAEMartins** [comment]
**@payam-zahedi and @HosamHasanRamadan**,  
I hope this message finds you both well.

Over the past few days, I've been working on the Stacked Expandable concept. Attached to this comment is a video showing my progress:  
[[Video link](https://github.com/user-attachments/assets/2b132f53-a749-4cc0-a6f0-a3fd60e0bf57)](https://github.com/user-attachments/assets/2b132f53-a749-4cc0-a6f0-a3fd60e0bf57)  

### Main Idea Behind the Implementation

Below is the core implementation concept:

```dart
OverlayEntry _createOverlayEntry() {
   ValueNotifier<bool> _isHovering = ValueNotifier<bool>(false);
   
   return OverlayEntry(
      opaque: false,
      builder: (context) {
         return Align(
            alignment: alignment,
            child: MouseRegion(...),
         );
      },
   );
}
```

I'm using the following elements to position the Toasts and achieve the expected animations and behavior:  
- `Stack`
- `AnimatedPositioned`
- `AnimatedOpacity`
- `TweenAnimationBuilder`

### Current Issues

While the implementation is progressing, I’m encountering some problems:  

1. **Overlay occupies entire vertical space:** When I hover over areas that don’t contain the Toast but are positioned below it, the Toasts expand unexpectedly.
2. **No exit animation:** The Toast currently lacks an exit animation.
3. **Hover behavior:** When hovering over the Toasts, only the hovered Toast has its expiration time frozen, while the others continue to expire.
4. **Dynamic updates:** To ensure the `Stack` updates correctly when new items are added, I had to add a `ValueNotifier<List<ToastificationItem>>` to the `_notifications` list.

**Comment by payam-zahedi** [comment]
Wow really nice 

Will you make a PR for that?

**Comment by AntonioAEMartins** [comment]
@payam-zahedi I've improved the code quality, but some of the issues are still present:

## Current Issues

1. Overlay occupies entire vertical space: When I hover over areas that don’t contain the Toast but are positioned below it, the Toasts expand unexpectedly.
2. No exit animation: The Toast currently lacks an exit animation.
3. Dynamic updates: To ensure the Stack updates correctly when new items are added, I had to add a ValueNotifier<List<ToastificationItem>> to the _notifications list.

**Comment by KiddoV** [comment]
Any update on this?

**Comment by AntonioAEMartins** [comment]
What's up @KiddoV 

Currently I'm really busy developing my Startup app/platform, so I'm not working in the Stacked Expandable Notifications.

If you want to help, I can share my thoughts around what I was developing. And together we can finish this PR

**Comment by KiddoV** [comment]
@AntonioAEMartins Please share so anyone can finish it!
